lib/xcodeproj/project/project_helper.rb

```
@@ -64,7 +64,60 @@ def self.new_target(project, type, name, platform, deployment_target, product_gr
 
         target
       end
+      #-----------------------------------------------------------------------#
+
+      # Duplicates a target and adds it to the project.
+      # @param  [Project] project
+      #         the project from which the target should be duplicated.
+      #
+      # @param  [klass] src_target
+      #         the target or the name of the target to duplicate
+      #
+      #
+      # @param  [String] dst_target_name
+      #         the name of the duplicated target.
+      #
+      #
+      # @return [PBXNativeTarget] the target.
+      #
+      def self.duplicate_target(project, src_target, dst_target_name)
+        if src_target.is_a?(String)
+          possible_targets = project.targets.select { |target| target.name.eql?(src_target) }
+          return nil if possible_targets.count == 0
+          # Currently only select the first one
+          target_to_duplicate = possible_targets.first
```

Or rather raise an exception instead?